Assets/Scripts/BlackHole.cs

```diff
@@ -21,31 +20,28 @@ void Update()
         
     }
 
-    private void OnTriggerEnter(Collider other)
+    private void OnTriggerStay(Collider other)
     {
-        moveCoroutine=StartCoroutine(MoveTowards(other));
-    }
-
-    private void OnTriggerExit(Collider other)
-    {
-        StopCoroutine(moveCoroutine);
-    }
-
-    IEnumerator MoveTowards(Collider other)
-    {
-        while (other!=null&&Vector3.Distance(other.transform.position, transform.position) > 50f)

```

Comment:
```suggestion
        if (other == null) return;
        while (Vector3.Distance(other.transform.position, transform.position) > 50f)
```
Vermeide redundanten Code, tu das `other != null` in allen weiteren Bedingungen weg.